### PR TITLE
test: wrap mocha.it to support to inject test plan id

### PR DESCRIPTION
**Pull request by LongOddCode**

### CODE REVIEW

File: packages/extra-shot-mocha/src/it.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  if (ctxOrFn) {
+    if (ctxOrFn instanceof Function) {
+      t = mocha.it(title, ctxOrFn);
+    } else {
+      t = mocha.it(title, async function (this) {
+        addContext(this, JSON.stringify(ctxOrFn));
+        await (fn as mocha.AsyncFunc).call(this);
+      });
+    }
+  } else {
```
Comment (1yefuwang1) [inline packages/extra-shot-mocha/src/it.ts:83]: It seems that the implementation doesn't handle edge cases like `it("should work", { caseId: 1 })`, which type-checks but throws at runtime.

**Review by 1yefuwang1 — DISMISSED** [review]

File: packages/extra-shot-mocha/tests/it.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  it("should run as normal mocha.it with sync arrow function", () => {
+    expect(1).equals(1);
+  });
+
+  it("should run as normal mocha.it with async arrow function", async () => {
+    expect(1).equals(1);
+  });
+
+  it("should inject ctx with async function", { a: 1 }, async function () {
+    expect(1).equals(1);
```
Comment (HuihuiWu-Microsoft) [inline packages/extra-shot-mocha/tests/it.test.ts:28]: Do we actually know whether the ctx is injected here?

Comment (LongOddCode) [reply packages/extra-shot-mocha/tests/it.test.ts:28]: it relies on the reporter. Currently, only mochawesome can output the context. See https://www.npmjs.com/package/mochawesome

Comment (LongOddCode) [reply packages/extra-shot-mocha/tests/it.test.ts:28]: image

Comment (HuihuiWu-Microsoft) [reply packages/extra-shot-mocha/tests/it.test.ts:28]: Maybe check the injection when available.

**Review by HuihuiWu-Microsoft — APPROVED** [review]
